bin/ruby-build

```diff
@@ -244,24 +244,33 @@ compute_md5() {
   fi
 }
 
+declare -a HAS_CHECKSUM_SUPPORT
+has_checksum_support() {
+  local checksum_command="$1"
+  return ${HAS_CHECKSUM_SUPPORT[$checksum_command]:=$(echo test | "$checksum_command" >/dev/null; echo $?)}
```

Comment:
Used indirection as suggested, but opted for `printf -v` to assign the variable instead of `eval`. I'm allergic to eval.